src/Component.js

```diff
@@ -46,6 +46,9 @@ export const DebounceInput = React.createClass({
 
 
   componentWillReceiveProps({value, debounceTimeout}) {
+    if (this.isDebouncing) {
```

Comment:
Nah, let it be the default behaviour. If it breaks anything, I'll just publish a new major version instad.